sixtyfps_runtime/interpreter/eval_layout.rs

```diff
@@ -0,0 +1,336 @@
+use sixtyfps_corelib::model::RepeatedComponent;
+use sixtyfps_corelib::slice::Slice;
+use sixtyfps_corelib::window::ComponentWindow;
+use std::convert::TryInto;
+
+pub(crate) fn compute_layout_info(lay: &Layout, local_context: &mut EvalLocalContext) -> Value {
+    let component = match local_context.component_instance {
+        ComponentInstance::InstanceRef(c) => c,
+        ComponentInstance::GlobalComponent(_) => panic!("Cannot compute layout from a Global"),
+    };
+    let expr_eval = |nr: &NamedReference| -> f32 {
+        eval::load_property(component, &nr.element(), nr.name()).unwrap().try_into().unwrap()
+    };
+    match lay {
+        Layout::GridLayout(grid_layout) => {
+            let cells = grid_layout_data(grid_layout, component, &expr_eval);
+            let (padding, spacing) = padding_and_spacing(&grid_layout.geometry, &expr_eval);
+            core_layout::grid_layout_info(Slice::from(cells.as_slice()), spacing, &padding).into()
+        }
+        Layout::BoxLayout(box_layout) => {
+            let (cells, alignment) = box_layout_data(box_layout, component, &expr_eval, None);
+            let (padding, spacing) = padding_and_spacing(&box_layout.geometry, &expr_eval);
+            core_layout::box_layout_info(
+                Slice::from(cells.as_slice()),
+                spacing,
+                &padding,
+                alignment,
+                box_layout.is_horizontal,
+            )
+            .into()
+        }
+        Layout::PathLayout(_) => unimplemented!(),
+    }
+}
+
+pub(crate) fn solve_layout(lay: &Layout, local_context: &mut EvalLocalContext) -> Value {
+    let component = match local_context.component_instance {
+        ComponentInstance::InstanceRef(c) => c,
+        ComponentInstance::GlobalComponent(_) => panic!("Cannot compute layout from a Global"),
+    };
+    let expr_eval = |nr: &NamedReference| -> f32 {
+        eval::load_property(component, &nr.element(), nr.name()).unwrap().try_into().unwrap()
+    };
+
+    match lay {
+        Layout::GridLayout(grid_layout) => {
+            let cells = grid_layout_data(grid_layout, component, &expr_eval);
+            let (padding, spacing) = padding_and_spacing(&grid_layout.geometry, &expr_eval);
+
+            Value::LayoutCache(core_layout::solve_grid_layout(&core_layout::GridLayoutData {
+                width: grid_layout
+                    .geometry
+                    .rect
+                    .width_reference
+                    .as_ref()
+                    .map(expr_eval)
+                    .unwrap_or(0.),
+                height: grid_layout
+                    .geometry
+                    .rect
+                    .height_reference
+                    .as_ref()
+                    .map(expr_eval)
+                    .unwrap_or(0.),
+                x: 0.,
+                y: 0.,
+                spacing,
+                padding: &padding,
+                cells: Slice::from(cells.as_slice()),
+            }))
```

Comment:
<nitpick>For consistency I think using `.into()` here instead of `Value::LayoutCache(...` is nicer :-)</nitpick>